pkg/version/plugin2/commonSpecForVCH.go

```diff
@@ -0,0 +1,128 @@
+	version = 2
+	target  = manager.ApplianceConfigure
+)
+
+func init() {
+	defer trace.End(trace.Begin(fmt.Sprintf("Registering plugin %s:%d", target, version)))
+	if err := manager.Migrator.Register(version, target, &AddCommonSpecForVCH{}); err != nil {
+		log.Errorf("Failed to register plugin %s:%d, %s", target, version, err)
+	}
+}
+
+// AddCommonSpecForVM is plugin for vic 0.8.0-GA version upgrade
+type AddCommonSpecForVCH struct {
+}
+
+type VirtualContainerHostConfigSpec struct {
+	ExecutorConfig `vic:"0.1" scope:"read-only" key:"init"`
+}
+
+type ExecutorConfig struct {
+	Common `vic:"0.1" scope:"read-only" key:"common"`
+}
+
+type Common struct {
+	// A reference to the components hosting execution environment, if any
+	ExecutionEnvironment string
+
+	// Unambiguous ID with meaning in the context of its hosting execution environment
+	ID string `vic:"0.1" scope:"read-only" key:"id"`
+
+	// Convenience field to record a human readable name
+	Name string `vic:"0.1" scope:"read-only" key:"name"`
+
+	// Freeform notes related to the entity
+	Notes string `vic:"0.1" scope:"hidden" key:"notes"`
+}
+
+type UpdatedVCHConfigSpec struct {
+	UpdatedExecutorConfig `vic:"0.1" scope:"read-only" key:"init"`
+}
+
+type UpdatedExecutorConfig struct {
+	UpdatedCommon `vic:"0.1" scope:"read-only" key:"common"`
+}
+
+type UpdatedCommon struct {
+	// A reference to the components hosting execution environment, if any
+	ExecutionEnvironment string
+
+	// Unambiguous ID with meaning in the context of its hosting execution environment
+	ID string `vic:"0.1" scope:"read-only" key:"id"`
+
+	// Convenience field to record a human readable name
+	Name string `vic:"0.1" scope:"hidden" key:"name"`
+
+	// Freeform notes related to the entity
+	Notes string `vic:"0.1" scope:"hidden" key:"notes"`
+}
+
+func (p *AddCommonSpecForVCH) Migrate(ctx context.Context, s *session.Session, data interface{}) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("AddCommonSpecForVCH version: %d", version)))
+	if data == nil {
+		return nil
+	}
+	mapData, ok := data.(map[string]string)
+	if !ok {
+		// Log the error here and return nil so that other plugins can proceed
+		log.Errorf("Migration data format is not map: %+v", data)
+		return nil
+	}
+	oldStruct := &VirtualContainerHostConfigSpec{}
+	result := extraconfig.Decode(extraconfig.MapSource(mapData), oldStruct)
+	log.Debugf("The oldStruct is %+v", oldStruct)
+	if result == nil {
+		return &errors.DecodeError{Err: fmt.Errorf("decode oldStruct %+v failed", oldStruct)}
+	}
+
+	newStruct := &UpdatedVCHConfigSpec{
+		UpdatedExecutorConfig: UpdatedExecutorConfig{
+			UpdatedCommon: UpdatedCommon{
+				Name:                 oldStruct.Name,
+				ID:                   oldStruct.ID,
+				ExecutionEnvironment: oldStruct.ExecutionEnvironment,
+				Notes:                oldStruct.Notes,
+			},
+		},
+	}
+
+	cfg := make(map[string]string)
+	extraconfig.Encode(extraconfig.MapSink(cfg), newStruct)
+	log.Debugf("The newStruct is %+v", newStruct)
```

Comment:
feels like the log is duplicated with https://github.com/vmware/vic/pull/4134/files#diff-b6b8049fb355ddb0278cc831df513dcdR124